Description: Build Qualcomm mink specific libckqteec library
Author: Abhinaba Rakshit <abhinaba.rakshit@oss.qualcomm.com>
Origin: other, Qualcomm forked mink specific changes over OPTEE-Client
Last-Update: 2026-07-14
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: optee-client/CMakeLists.txt
===================================================================
--- optee-client.orig/CMakeLists.txt
+++ optee-client/CMakeLists.txt
@@ -1,11 +1,9 @@
-cmake_minimum_required(VERSION 3.4)
-project(optee_client C)
-
-# https://cmake.org/Wiki/CMake_Useful_Variables
-set(CMAKE_TOOLCHAIN_FILE CMakeToolchain.txt)
+cmake_minimum_required(VERSION 3.10)
+project(ckqteec
+	VERSION 0.1.0
+	LANGUAGES C)
 
 option(CFG_WERROR "Build with -Werror" TRUE)
-option(WITH_TEEACL "Build libteeacl" TRUE)
 
 include(GNUInstallDirs)
 
@@ -22,6 +20,7 @@ add_compile_options(
 	-Wmissing-noreturn -Wmissing-prototypes -Wnested-externs
 	-Wpointer-arith -Wshadow -Wstrict-prototypes
 	-Wswitch-default -Wwrite-strings -fPIC
+	-Wno-unterminated-string-initialization
 )
 if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
     add_compile_options(
@@ -38,12 +37,75 @@ if(CCACHE_FOUND)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif(CCACHE_FOUND)
 
-add_subdirectory(libteec)
-add_subdirectory(tee-supplicant)
-add_subdirectory(libckteec)
-if(WITH_TEEACL)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
-    add_subdirectory(libteeacl)
-endif(WITH_TEEACL)
-add_subdirectory(libseteec)
+################################################################################
+# Packages
+################################################################################
+find_package(Threads REQUIRED)
+if(NOT THREADS_FOUND)
+	message(FATAL_ERROR "Threads not found")
+endif()
+
+include(GNUInstallDirs)
+
+################################################################################
+# Source files
+################################################################################
+set(SRC
+	libckteec/src/pkcs11_api.c
+	libckteec/src/ck_debug.c
+	libckteec/src/ck_helpers.c
+	libckteec/src/invoke_ta.c
+	libckteec/src/pkcs11_processing.c
+	libckteec/src/pkcs11_token.c
+	libckteec/src/serializer.c
+	libckteec/src/serialize_ck.c
+	src/teec_trace.c
+)
+
+################################################################################
+# Built library
+################################################################################
+add_library(ckqteec SHARED ${SRC})
+
+set_target_properties(ckqteec PROPERTIES
+	VERSION ${PROJECT_VERSION}
+	SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+################################################################################
+# Flags always set
+################################################################################
+target_compile_definitions(ckqteec
+	PRIVATE -D_GNU_SOURCE
+	PRIVATE -DBINARY_PREFIX="LT"
+)
+
+################################################################################
+# Optional flags
+################################################################################
+
+################################################################################
+# Public and private header and library dependencies
+################################################################################
+target_include_directories(ckqteec
+	PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libckteec/include>
+	$<INSTALL_INTERFACE:libckteec/include>
+	PRIVATE libckteec/src
+	PRIVATE include
+)
+
+target_link_libraries(ckqteec
+	PRIVATE pthread
+	PRIVATE minkteec
+	PRIVATE minkadaptor
+	PRIVATE m
+)
+
+################################################################################
+# Install targets
+################################################################################
+install(TARGETS ckqteec
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+add_subdirectory(libckteec/include)
